shared/lib_shared_inverter.cpp

```diff
@@ -375,7 +375,10 @@ void SharedInverter::convertOutputsToKWandScale(double tempLoss, double powerAC_
     powerConsumptionLoss_kW *= m_numInverters * util::watt_to_kilowatt;
     powerNightLoss_kW *= m_numInverters * util::watt_to_kilowatt;
     powerTempLoss_kW = tempLoss * m_numInverters * util::watt_to_kilowatt;
-    powerLossTotal_kW = powerDC_kW - powerAC_kW;
+    if (powerDC_kW < 0.0)
+        powerLossTotal_kW = fabs(powerDC_kW) - fabs(powerAC_kW); // DC connected grid charging
```

Comment:
Are the charging and discharging losses all accounted for in the difference here?

Reply:
This is the fix for https://github.com/NREL/ssc/issues/631

The more detailed loss accounting is in lib_battery_power_dispatch.cpp, scattered throughout the calculation functions. There's also code in cmod_battery that does updates for the actual AC losses in pvsamv1. Many of the changes in those three files are related to loss computation.